File: rails/app/models/draft_participant.rb
class DraftParticipant < ApplicationRecord
  self.table_name = 'draft_participants'

  has_many :picks, class_name: 'DraftPick', inverse_of: :participant
  belongs_to :session, class_name: 'DraftSession', inverse_of: :participants
  belongs_to :player, class_name: 'Player', inverse_of: :draft_sessions

  # Domain invariants — simple rules
  validate :validate_rules

  def validate_rules
    errors.add(:seat_number_positive, 'Seat number must be greater than zero') unless ((seat_number.nil? || seat_number > 0))
  end

  def to_s
    seat_number.to_s
  end

  # Business operations

  def pick_card(card_id, pack_number)
    # TODO: implement pick_card
  end

  def drafted_card_count
    # TODO: implement drafted_card_count
    nil
  end

  def as_json(options = {})
    hash = super(options)
    hash['joinedAt'] = hash.delete('joined_at') if hash.key?('joined_at')
    hash
  end
end

File: dotnet/CardsProject/Domain/Content/DraftParticipant.cs
using CardsProject.Domain.Players;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace CardsProject.Domain.Content;

public class DraftParticipant : IValidatableObject
{
    public int Id { get; set; }

    public int SeatNumber { get; set; } = 0;
    [JsonPropertyName("joinedAt")]
    public DateTime? JoinedAt { get; set; } = null;

    public int? SessionId { get; set; }
    [ForeignKey(nameof(SessionId))]
    public DraftSession? Session { get; set; }
    public int? PlayerId { get; set; }
    [ForeignKey(nameof(PlayerId))]
    public Player? Player { get; set; }

    // Business operations

    public void PickCard(int cardId, int packNumber)
    {
        // TODO: implement pick_card
    }

    public int DraftedCardCount()
    {
        // TODO: implement drafted_card_count
        return default;
    }

    // ── Domain invariants (simple rules) ──────────────────────────────
    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        if (!( SeatNumber > 0 ))
            yield return new ValidationResult("Seat number must be greater than zero", new[] { nameof(Id) });
    }
}
